audiovario.c

```diff
@@ -76,58 +77,62 @@ float change_volume(float delta){
   return volume;
 }
 
-float pulse_syn(float phase, float rise, float fall, float duty ){
+inline float pulse_syn(float phase, float rise, float fall, float risei, float falli, float duty ){
   float ret;
 
-  phase= fmodf(phase,2*m_pi);
-
   if (phase <rise) {
-    ret=phase/rise;
+    ret=phase*risei;
     return ret>1.0?1.0:ret;
   }
-  else if (phase < rise+duty) return 1.0;
-  else if (phase < rise+duty+fall) {
-    ret=1.0-((phase-rise-duty)/fall);
+  if (phase < rise+duty) return 1.0;
+  if (phase < rise+duty+fall) {
+    ret=1.0-((phase-rise-duty)*falli);
```

Comment:
What this change is for?

Reply:
See above.  Multiply by the reciprocal instead of dividing.  I was hoping it was a speed improvement.